plugins/woocommerce/src/Blocks/BlockTypes/ProductGalleryThumbnails.php

```diff
@@ -119,7 +119,7 @@ protected function should_display_view_all( $mode, $thumbnails_count, $product_g
 	 * @return string Rendered block type output.
```

Comment:
This does not always return a `string` type. We either should do, or update this

Reply:
Its the same case with the `inject_view_all` method

Reply:
When defining `$number_of_thumbnails` we assume `$block->context['thumbnailsNumberOfThumbnails']` is a number but we don't explicitly check this. We probably should also check its an expected data type

Reply:
I don't think we sanitize the `$html` before returning it. We could use `wp_kses_post` or `esc_html` Im not sure which is a better fit here? Probably the former?

Reply:
Nice ones!

> This does not always return a string type. We either should do, or update this

Fixed

> Its the same case with the inject_view_all method

Fixed

> When defining $number_of_thumbnails we assume $block->context['thumbnailsNumberOfThumbnails'] is a number but we don't explicitly check this. We probably should also check its an expected data type

Fixed

> I don't think we sanitize the $html before returning it. We could use wp_kses_post or esc_html Im not sure which is a better fit here? Probably the former?

Yeah, I used `wp_kses_post`. We expect HTML there. Good catch!

https://github.com/woocommerce/woocommerce/pull/54682/commits/6eb2122ff512f4f5f6a8d89ddc39c79a43ec102f